lib/checks/aria/required-children.js

```diff
@@ -1,58 +1,87 @@
-const requiredOwned = axe.commons.aria.requiredOwned;
-const implicitNodes = axe.commons.aria.implicitNodes;
-const matchesSelector = axe.utils.matchesSelector;
-const idrefs = axe.commons.dom.idrefs;
-const hasContentVirtual = axe.commons.dom.hasContentVirtual;
+const { aria, dom } = axe.commons;
+const { requiredOwned, implicitNodes, getRole } = aria;
+const { hasContentVirtual, idrefs } = dom;
+const { matchesSelector, querySelectorAll, getNodeFromTree } = axe.utils;
+
 const reviewEmpty =
 	options && Array.isArray(options.reviewEmpty) ? options.reviewEmpty : [];
+const role = node.getAttribute('role');
+const required = requiredOwned(role);
+if (!required) {
+	return true;
+}
 
-function owns(node, virtualTree, role, ariaOwned) {
-	if (node === null) {
-		return false;
-	}
-	const implicit = implicitNodes(role);
-	let selector = ['[role="' + role + '"]'];
-
-	if (implicit) {
-		selector = selector.concat(
-			implicit.map(implicitSelector => implicitSelector + ':not([role])')
-		);
-	}
+let all = false;
+let childRoles = required.one;
+if (!childRoles) {
+	all = true;
+	childRoles = required.all;
+}
 
-	selector = selector.join(',');
-	return ariaOwned
-		? matchesSelector(node, selector) ||
-				!!axe.utils.querySelectorAll(virtualTree, selector)[0]
-		: !!axe.utils.querySelectorAll(virtualTree, selector)[0];
+const ownedElements = idrefs(node, 'aria-owns');
+const descendantRole = getDescendantRole(node, ownedElements);
+const missing = missingRequiredChildren(
```

Comment:
Instead of upping the eslint `max-params` limit you could pass all these as a single object and destructure it.

```js
missingRequiredChildren({
  node,
  childRoles,
  all,
  role,
  ownedElements,
  descendantRole
});

function missingRequiredChildren({ 
  node,
  childRoles,
  all,
  role,
  ownedElements,
  descendantRole
}) {
  // ...
}
```